src/target/cortexm.c

```diff
@@ -386,7 +401,10 @@ void cortexm_detach(target *t)
 	/* Clear any stale watchpoints */
 	for(i = 0; i < priv->hw_watchpoint_max; i++)
 		target_mem_write32(t, CORTEXM_DWT_FUNC(i), 0);
-
+	/* Release exception catch .*/
+	target_mem_write32(t, CORTEXM_DEMCR, 0);
```

Comment:
I don't know why you added this.  These DEMCR flags have no effect when halting debug is disabled.
Target execution would be resumed by the write to DHCSR below; there is no need to call `cortexm_halt_resume` explicitly.